biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java

```diff
@@ -511,13 +512,20 @@ private void loadPreferences() {
 		resolvePrefs = new Parameters(properties.getProperty(PROP_RESOLVE_PREFERENCES), project);
 	}
 
+	private List<ResolverHook> getResolverHooks() {
+		if (resolverHooks != null) {
+			return resolverHooks;
+		}
+		return resolverHooks = registry.getPlugins(ResolverHook.class);
+	}
+
 	@Override
 	protected void postProcessProviders(Requirement requirement, Set<Capability> wired, List<Capability> candidates) {
 		if (candidates.isEmpty())
 			return;
 
 		// Call resolver hooks
-		for (ResolverHook resolverHook : registry.getPlugins(ResolverHook.class)) {
+		for (ResolverHook resolverHook : getResolverHooks()) {
```

Comment:
I am not sure I understand the value of this change. Is this necessary for correctness? Or is is some optimization? It seems the safe thing is to always get the current hooks rather than only get them once.

Reply:
This was an optimization. I noticed while debugging that the code was passing through the plugin calculation logic (which is not trivial) to find resolve hooks about a billion times.
Ask yourself if it makes sense if, once a resolve has already started, a new resolve hook should suddenly appear. Wouldn't that wreak havoc over the result having had only a subset be subject to any resolve hooks?

Reply:
Of minor consequence, it also sped up the build a bit, but enough to notice.

Reply:
OK, It just wasn't clear to me whether this was part of the bug fix or just some optimization you tucked into this PR. It is the latter.